### ISSUE TITLE
Affiliated packages should be able to run tests in IPython if they wish

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#3184]
At the moment, running tests in IPython gives:

```
/Users/tom/Library/Python/3.4/lib/python/site-packages/astropy-1.0.dev10666-py3.4-macosx-10.8-x86_64.egg/astropy/tests/helper.py in run_tests(self, package, test_path, args, plugins, verbose, pastebin, remote_data, pep8, pdb, coverage, open_files, parallel, docs_path, skip_docs, repeat)
    117         else:
    118             raise RuntimeError(
--> 119                 "Running astropy tests inside of IPython is not supported.")
    120 
    121         if coverage:

RuntimeError: Running astropy tests inside of IPython is not supported.
```

even for affiliated packages. We disabled this in Astropy core because of some tests not interacting well with IPython but there's no reason we should prevent affiliated packages from doing this.

**Comment by bsipocz** [comment]
@embray - I think this would be great to fix in 1.0.2, either by merging the hack in  #3254 or xfailing the failing tests if any (I would volunteer to do the latter too, but I have no test failing, only some errors coming from the nddata tests that look unrelated...)

**Comment by embray** [comment]
@bsipocz I'll have a look.

**Comment by embray** [comment]
Does anybody know what issues used to fail in IPython?  Because I just ran `astropy.test()` in IPython and didn't get any failures at all.  Were they non-deterministic?

**Comment by embray** [comment]
Note, this was with IPython 2.3.1.  YMMV with older versions.

**Comment by mhvk** [comment]
@embray - I've been meaning to ask about this, since I rather liked testing inside ipython, and never had problems until it became disallowed (but so far I hadn't tried to circumvent this -- how do you run the tests in ipython?).

**Comment by embray** [comment]
I just removed the code that prevented it.  Now I need to go back and see if I can figure out what the problem used to be in.

**Comment by embray** [comment]
The only thing even remotely concrete I've been able to find is running into problems with were `sys.meta_path` loaders installed sometimes by IPython.  Though that's not a problem limited to IPython, and if there is such a problem that should be fixed.

**Comment by embray** [comment]
If not `sys.meta_path`, I think it may also have been that there were configuration-dependent failures that were possible, but I think now we're better about running the tests against a consistent configuration.

**Comment by embray** [comment]
This should be addressed, for now, by #3678.
